### PR TITLE
[BD-14] feat: restyle create new library page

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:**
- Upgrade the` @edx/paragon` version to the latest available versions for current packages.
- Refactor the "create a new library" popup on a separate page.
- Replace all components with paragon components.
- Replace bootstrap styles with paragon styles.
- Add the ability to add custom branding via the `@edx/brand` component.
- Add a modal which shows a message about not saving data

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ihor-romaniuk!


When this pull request is ready, tag your edX technical lead.

**Comment by ihor-romaniuk** [comment]
@connorhaugh Please, take a look at this PR.

File: src/library-authoring/create-library/LibraryCreatePage.jsx [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+              <h2 className="page-header h2">{intl.formatMessage(messages['library.form.create.library'])}</h2>
+            </header>
+          </div>
+          <div className="wrapper-content wrapper">
+            <section className="content">
+              <form onSubmit={this.onSubmit} className="form-create">
+                <fieldset>
+                  <ol className="list-input">
+                    <li className="field">
+                      <FormGroup
```
Comment (connorhaugh) [inline src/library-authoring/create-library/LibraryCreatePage.jsx:231]: I am a little confused as to why your `FormGroup` custom component only includes one form control. In its normal uses, a form group controls multiple form.control or other input components.

File: src/library-authoring/common/FormGroup.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  const handleClick = (e) => {
+    if (props.handleClick) { props.handleClick(e); }
+  };
+  const handleOnBlur = (e) => {
+    if (props.handleBlur) { props.handleBlur(e); }
+  };
+
+  return (
+    <Form.Group isInvalid={!!props.errorMessage} className={props.className}>
+      <Form.Control
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/library-authoring/common/FormGroup.jsx:19]: Thanks for your advice, I simplified props by using `{...props}`. 

And what about using {props.children}, it is already used by the parent `Form.Group` component.

File: src/library-authoring/create-library/LibraryCreatePage.jsx (hunk 1 above)
Comment (connorhaugh) [reply src/library-authoring/create-library/LibraryCreatePage.jsx:231]: Yeah, that makes sense as to why it was used, and I guess it is in fact a wrapper for a formgroup. Sometimes a "better name" is more trouble than it is worth.

File: package.json [hunk 3]
```diff
@@ -55,6 +56,7 @@
     "react": "17.0.2",
     "react-dom": "17.0.2",
     "react-dropzone": "12.0.4",
+    "react-onclickoutside": "^6.12.1",
```
Comment (connorhaugh) [inline package.json:59]: just to note for anyone else who looks:  this new library is very widely used. (https://www.npmjs.com/package/react-onclickoutside)

File: src/library-authoring/create-library/LibraryCreatePage.jsx (hunk 1 above)
Comment (ihor-romaniuk) [reply src/library-authoring/create-library/LibraryCreatePage.jsx:231]: Yes, you are definitely right, it's just a wrapper for a FormGroup component.
Is it better to rename it like FormGroupWrapper or something like that?

**Review by connorhaugh — APPROVED** [review]
Looks good to me based on the assumption that this passes the sandbox review.

File: src/library-authoring/create-library/LibraryCreatePage.jsx (hunk 1 above)
Comment (connorhaugh) [reply src/library-authoring/create-library/LibraryCreatePage.jsx:231]: I wouldn't worry too much about it.

**Comment by connorhaugh** [comment]
Looks like some dependancies are out of date-- can you resolve them?

**Comment by ihor-romaniuk** [comment]
> Looks like some dependancies are out of date-- can you resolve them?

It happened because of the high version of React. The footer component has peer dependencies for React `^16.9.0`.
I'm not sure that downgrading react version it's a good idea.

What do you think of this and what do you think of the best way to solve it?

**Comment by connorhaugh** [comment]
https://github.com/openedx/frontend-component-footer/commit/d4259ef1f6923f1b739ad2d07199027ab87b42d5 looks like the react version was pinned three years ago for no reason, so we can probably make a PR to unpin it.

**Pull request merged**